pytorch_pfn_extras/training/extensions/progress_bar_notebook.py

```diff
@@ -0,0 +1,140 @@
+import datetime
+import sys
+import time
+
+from IPython.core.display import display
+from ipywidgets import HTML, FloatProgress, HBox, VBox
+
+from pytorch_pfn_extras.training import extension, trigger
+
+
+class ProgressBarNotebook(extension.Extension):
+
+    """An extension to print a progress bar and recent training updater.
+
+    This extension is aimed to work on jupyter notebook as replacement of `ProgressBar`.
+    This extension prints a progress bar at every call. It watches the current
+    iteration and epoch to print the bar.
+
+    Args:
+        training_length (tuple or None): Length of whole training. It consists
+            of an integer and either ``'epoch'`` or ``'iteration'``. If this
+            value is omitted and the stop trigger of the manager is
+            :class:`IntervalTrigger`, this extension uses its attributes to
+            determine the length of the training.
+        update_interval (int): Number of iterations to skip printing the
+            progress bar.
+        bar_length (int): This is not used, argument is kept to be consistent with `ProgressBar`.
+        out: This is not used, argument is kept to be consistent with `ProgressBar`.
+    """
+
+    def __init__(self, training_length=None, update_interval=100,
+                 bar_length=50, out=sys.stdout):
```

Comment:
I kept unused 2 arguments to switch between ProgressBar & ProgressBarNotebook easily. what do you think?